# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1457490713 18000
#      Tue Mar 08 21:31:53 2016 -0500
# Node ID 0ad3fe00d6f9083bae3c697b9f706d66e67741ba
# Parent  d6c6cf748fc9553575c03371058fa8e6397ad9df
Consistent linestyles for gnuplot terminals (Bug # 46886).

__gnuplot_draw_axes__.m (gnuplot_linetype): When supported use the gnuplot
dashtype feature to specify Octave's linestyles.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1938,23 +1938,23 @@ function [style, ltidx] = do_linestyle_c
 endfunction
 
 function [lt] = gnuplot_linetype (obj)
   if (isfield (obj, "linestyle"))
     if (__gnuplot_has_feature__ ("dashtype"))
       opt = "dashtype";
       switch (obj.linestyle)
         case "-"
-          lt = "1";
+          lt = "solid";
         case "--"
-          lt = "2";
+          lt = "'_ '";
         case ":"
-          lt = "3";
+          lt = "'. '";
         case "-."
-          lt = "4";
+          lt = "'-. '";
         case "none"
           lt = "";
         otherwise
           lt = "";
       endswitch
     elseif (__gnuplot_has_feature__ ("linetype"))
       opt = "linetype";
       switch (obj.linestyle)
